libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -713,6 +717,37 @@ def test_similarity_search_with_relevance_score(
     assert len(result) == number_of_docs_to_return
 
 
+def test_similarity_search_with_relevance_score_result_constraint(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+    metadatas: List[dict],
+) -> None:
+    """Test that the result from similarity_search_with_relevance_score
+    is capped between 0 and 1."""
+    store.add_texts(texts, metadatas)
+    result = store.similarity_search_with_relevance_scores("Good review")
+
+    for value in result:
+        assert (
+            value[1] >= 0 and value[1] <= 1
+        ), f"Relevance score {value[1]} not in range [0, 1]."
+
+
+def test_select_relevance_score_fn_returns_correct_relevance_function(
+    store: SQLServer_VectorStore,
+    docs: List[Document],
+) -> None:
+    """Test that `_select_relevance_score_fn` uses the appropriate
+    relevance function based on distance strategy provided."""
+    store.add_documents(docs)
+    result = store._select_relevance_score_fn()
+
+    # This vectorstore is initialized to use `COSINE` distance strategy,
+    # as such, the relevance function returned should be a
+    # cosine relevance function.
+    assert result == store._cosine_relevance_score_fn
```

Comment:
can we test if we give error if the distance strategy is not one of the supported ones?

Reply:
Added a test for that.